MewsUtils/Posts.py

```diff
@@ -178,28 +181,47 @@ def getRelatedPosts(pid, skip, amount):
     # Query Mews-App DB
     cursor = cnx.cursor(dictionary=True)
     query = '''
-        SELECT 
-            post1_id,
-            post2_id,
-            rel_txt_wt,
-            rel_txt_meta,
-            ocr_meta,
-            sub_img_wt,
-            ocr_wt,
-            ifnull(rel_txt_wt, 0) * %(rel_txt_wt)s + 
-            ifnull(sub_img_wt, 0) * %(sub_img_wt)s + 
-            ifnull(ocr_wt, 0) * %(ocr_wt)s 
-            as total_wt
+        SELECT
+            A.id, A.image_url,
+            A.post_url, A.reposts,
+            A.replies, A.likes,
+            A.when_posted, A.user_id,
+            A.related_text, A.ocr_text,
+            A.when_scraped, A.when_updated,
+            B.rel_txt_wt,
+            B.rel_txt_meta,
+            B.ocr_meta,
+            B.sub_img_wt,
+            B.ocr_wt,
+            B.total_wt
         FROM 
-            mews_app.PostRelatedness
+            mews_app.Posts AS A,
+            (SELECT 
+                post1_id,
+                post2_id,
+                IF(post1_id = %(post_id)s, post2_id, post1_id) AS rel_id,
+                rel_txt_wt,
+                rel_txt_meta,
+                ocr_meta,
+                sub_img_wt,
+                ocr_wt,
+                ifnull(rel_txt_wt, 0) * %(rel_txt_wt)s + 
+                ifnull(sub_img_wt, 0) * %(sub_img_wt)s + 
+                ifnull(ocr_wt, 0) * %(ocr_wt)s 
+                AS total_wt
```

Comment:
I've recently made total_wt a derived column of the PostRelatedness table, so you can actually just select it directly

Reply:
Just updated. Thanks for heads up.